metrics-exporter-prometheus/src/exporter/http_listener.rs

```diff
@@ -0,0 +1,131 @@
+use std::{
+    net::{IpAddr, SocketAddr},
+    sync::Arc,
+};
+
+use http_body_util::Full;
+use hyper::{
+    body::{self, Bytes, Incoming},
+    server::conn::http1::Builder as HyperHttpBuilder,
+    service::service_fn,
+    Request, Response, StatusCode,
+};
+use hyper_util::rt::TokioIo;
+use ipnet::IpNet;
+use tokio::net::{TcpListener, TcpStream};
+use tracing::warn;
+
+use crate::{common::BuildError, ExporterFuture, PrometheusHandle};
+
+struct HttpListeningExporter {
+    inner: Arc<Inner>,
+}
+#[derive(Clone)]
+struct Inner {
+    handle: PrometheusHandle,
+    allowed_addresses: Option<Vec<IpNet>>,
+}
+
+impl HttpListeningExporter {
+    async fn serve(&self, listener: std::net::TcpListener) -> Result<(), hyper::Error> {
+        // TODO - address panic possibility
+        let listener = TcpListener::from_std(listener).unwrap();
+
+        loop {
+            let stream = match listener.accept().await {
+                Ok((stream, _)) => stream,
+                Err(e) => {
+                    warn!("Error accepting connection. Ignoring request. Error: {:?}", e);
+                    continue;
+                }
+            };
+
+            let remote_addr = match stream.peer_addr() {
+                Ok(remote_address) => remote_address.ip(),
+                Err(e) => {
+                    warn!("Error obtaining remote address. Ignoring request. Error: {:?}", e);
```

Comment:
Add a dedicated `error` field on the event that we assign the error value to.

(see previous comments about logging for format example)